### ISSUE TITLE
TESTNET BUG - Crash after Opened TrieFileStorage

### ISSUE THREAD

**Issue by dantrevino** [blockstack/stacks-blockchain#1825]
**Describe the bug**
Miner crashes with thread panic

**Steps To Reproduce**
Run miner with argon setup

**Expected behavior**
Miner should continue to run

**Environment (please complete the following information):**
 - OS:Ubuntu 20.04 Arm64
 - Rust version: None ... using Binaries from this merge: https://github.com/blockstack/stacks-blockchain/actions/runs/209094967
 - Version of the appropriate binary / software package
`stacks-node 0.1.0 => 23.0.0.0 (master:2fb1dcf+, release build, linux [aarch64])`

**Additional context**
```
Aug 23 01:57:58 pi4miner stacks-node[2261]: DEBUG [1598147878.147] [testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:792] [ThreadId(1)] BitcoinRPC builder: None:None@http://35.236.218.197:18443/
Aug 23 01:57:59 pi4miner stacks-node[2261]: INFO [1598147879.247] [testnet/stacks-node/src/run_loop/neon.rs:94] [ThreadId(1)] Miner node: starting up, UTXOs found.
Aug 23 01:57:59 pi4miner stacks-node[2261]: DEBUG [1598147879.252] [src/chainstate/burn/db/sortdb.rs:1072] [ThreadId(1)] Connect/Open sortdb '/mnt/stacks/burnchain/db/bitcoin/regtest/sortition.db/data.db' as 'readwrite', with index as '/mnt/stacks>
Aug 23 01:57:59 pi4miner stacks-node[2261]: DEBUG [1598147879.253] [src/chainstate/stacks/index/storage.rs:661] [ThreadId(1)] Opened TrieFileStorage /mnt/stacks/burnchain/db/bitcoin/regtest/sortition.db/marf;
Aug 23 01:57:59 pi4miner stacks-node[2261]: Process abort due to thread panic
Aug 23 01:57:59 pi4miner stacks-node[2261]:    0: stacks_node::main::{{closure}}
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at src/testnet/stacks-node/src/main.rs:44
Aug 23 01:57:59 pi4miner stacks-node[2261]:    1: std::panicking::rust_panic_with_hook
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/panicking.rs:490
Aug 23 01:57:59 pi4miner stacks-node[2261]:    2: rust_begin_unwind
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/panicking.rs:388
Aug 23 01:57:59 pi4miner stacks-node[2261]:    3: std::panicking::begin_panic_fmt
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/panicking.rs:342
Aug 23 01:57:59 pi4miner stacks-node[2261]:    4: blockstack_lib::burnchains::burnchain::<impl blockstack_lib::burnchains::Burnchain>::sync_with_indexer
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/macros.rs:16
Aug 23 01:57:59 pi4miner stacks-node[2261]:    5: stacks_node::burnchains::bitcoin_regtest_controller::BitcoinRegtestController::receive_blocks
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at src/testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:150
Aug 23 01:57:59 pi4miner stacks-node[2261]:    6: <stacks_node::burnchains::bitcoin_regtest_controller::BitcoinRegtestController as stacks_node::burnchains::BurnchainController>::start
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at src/testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:571
Aug 23 01:57:59 pi4miner stacks-node[2261]:       stacks_node::run_loop::neon::RunLoop::start
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at src/testnet/stacks-node/src/run_loop/neon.rs:102
Aug 23 01:57:59 pi4miner stacks-node[2261]:    7: stacks_node::main
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at src/testnet/stacks-node/src/main.rs:96
Aug 23 01:57:59 pi4miner stacks-node[2261]:    8: std::rt::lang_start::{{closure}}
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/rt.rs:67
Aug 23 01:57:59 pi4miner stacks-node[2261]:    9: std::rt::lang_start_internal::{{closure}}
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/rt.rs:52
Aug 23 01:57:59 pi4miner stacks-node[2261]:       std::panicking::try::do_call
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/panicking.rs:297
Aug 23 01:57:59 pi4miner stacks-node[2261]:       std::panicking::try
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/panicking.rs:274
Aug 23 01:57:59 pi4miner stacks-node[2261]:       std::panic::catch_unwind
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/panic.rs:394
Aug 23 01:57:59 pi4miner stacks-node[2261]:       std::rt::lang_start_internal
Aug 23 01:57:59 pi4miner stacks-node[2261]:              at rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/rt.rs:51
Aug 23 01:57:59 pi4miner stacks-node[2261]:   10: main
Aug 23 01:57:59 pi4miner stacks-node[2261]:   11: __libc_start_main
Aug 23 01:57:59 pi4miner stacks-node[2261]:   12: <unknown>
Aug 23 01:57:59 pi4miner systemd[1]: stacks-node.service: Main process exited, code=exited, status=1/FAILURE
```

**Comment by timstackblock** [comment]
Thanks for the submission @dantrevino have you been able to reproduce this issue.

**Comment by timstackblock** [comment]
Seems to be a possible duplicate of https://github.com/blockstack/stacks-blockchain/issues/1795

**Comment by dantrevino** [comment]
I did see this multiple time, but my current node was running 2 days with no issues, before the restart today (Sep 2).

**Comment by jcnelson** [comment]
Addressed in #1876
